### ISSUE TITLE
Markdown assumes comment in code section as a big title.

### ISSUE THREAD

**Issue by ivo-at-shake** [microsoft/vscode#158179]
Does this issue occur when all extensions are disabled?: Not sure...

- VS Code Version:
- OS Version:

Steps to Reproduce:

1. Create two markdown cells, in a jupyter notebook. 
2. In the first cell, write # Big Title
3. In the second cell, write:

~~~md
This text
```
# should be inside the Big Title section, when collapsed.
```
But it's not.
~~~

4. Collapse the first cell, and see that the second cell, due to the # in the code part, does not collapse to its interior.

**Comment by PandaGnar** [comment]
The bug seems to be fixed on my end (see videos in PR).

I added some tests just to confirm in the PR above, but it's not too important whether that PR gets merged.

**Comment by mjbvz** [comment]
Thanks for the follow up. Yes this was possibly fixed by d736ca90ea389fd52097f71e9a1135342a8ff1ee
